server/types/interface.go

```diff
@@ -113,3 +113,36 @@ var typesFromBaseID = map[DoltgresTypeBaseID]DoltgresType{
 	Xid.BaseID():              Xid,
 	XidArray.BaseID():         XidArray,
 }
+
+// typesFromBaseID contains a map from a DoltgresTypeBaseID to its originating type.
+var typesFromTypName = map[string]DoltgresType{
+	"anyarray":    AnyArray,
+	"bpchar":      BpChar,
+	"bool":        Bool,
+	"bytea":       Bytea,
+	"char":        CharType{Length: 1},
+	"date":        Date,
+	"float4":      Float32,
+	"float8":      Float64,
+	"int2":        Int16,
+	"int4":        Int32,
+	"int8":        Int64,
+	"json":        Json,
+	"jsonb":       JsonB,
+	"name":        Name,
+	"numeric":     Numeric,
+	"oid":         Oid,
+	"text":        Text,
+	"time":        Time,
+	"timestamp":   Timestamp,
+	"timestamptz": TimestampTZ,
+	"timetz":      TimeTZ,
+	"uuid":        Uuid,
+	"unknown":     Unknown,
+	"varChar":     VarChar,
+	"xid":         Xid,
+}
+
+func GetAllPgTypes() map[string]DoltgresType {
+	return typesFromTypName
+}
```

Comment:
Rather than do this, we should instead add a `BaseName()` function on `DoltgresType` and get the name from there. We want to minimize the number of locations that need to be updated when adding a new type. We already have `typesFromBaseID` as a collection containing every type, so we should use that or construct other maps from it (which is what we do for `baseIDArrayTypes`).

Reply:
Also just noticed that you have `varChar`. Is the `C` supposed to be capitalized?